### DISCUSSION TITLE
[Rule Proposition] avoid_unassigned_late_fields

### DISCUSSION THREAD

**Discussion by TesteurManiak** [Floating-Dartists/fd_lints#4] [Ideas]
@Pierre-Monier @MeixDev what do you think of this rule?

**AVOID** late fields that are unassigned inside their own class's constructor or method.

**BAD:**
```dart
class Foo {
  late final int bar;
}
```

**GOOD:**
```dart
class Foo {
  late final int bar;

  Foo() {
    bar = 42;
  }
}
```

**GOOD:**
```dart
class Foo {
  late final int bar;

  void baz() {
    bar = 24;
  }
}
```

**GOOD:**
```dart
class Foo {
  late final int bar;

  Foo() : bar = 42;
}
```

**Comment by TesteurManiak** [comment]
You can check a first try here: https://github.com/Floating-Dartists/fd_lints/blob/feat/avoid-unassigned-late-fields/lib/src/lints/avoid_unassigned_late_fields.dart

**Comment by Pierre-Monier** [comment]
Great one, I can't review your custom lint code because I have no idea of how it works

**Reply by TesteurManiak** [reply]
Don't worry the idea was more, to know if you'd have any other test case in mind to ensure that there's no forgotten edge case 👍

**Comment by TesteurManiak** [comment]
Added in https://github.com/Floating-Dartists/fd_lints/pull/6

**Comment by Pierre-Monier** [comment]
LGTM
